proposed/http-message.md

```diff
@@ -1402,9 +1404,11 @@ interface UriInterface
      * This method MUST retain the state of the current instance, and return
      * an instance that contains the specified query string.
      *
-     * If the query string is prefixed by "?", that character MUST be removed.
-     * Additionally, the query string SHOULD be parseable by parse_str() in
-     * order to be valid.
+     * The query string SHOULD be parseable by parse_str() in order to be
```

Comment:
This is why it's a SHOULD and not a MUST. Typically, for PHP, if it's not parseable by `parse_str()`, `$_GET`, and thus `getQueryParams()`, will be empty. However, since we are also targeting client-side usage, I think we can safely remove it.